code/io/keycontrol.cpp

```diff
@@ -1448,6 +1453,10 @@ void game_do_end_mission_popup()
 			break;
 
 		case 2:
+			// Cyborg17 - best place to check for this *one* looping sound.
```

Comment:
As I mentioned in #1433, this should be near the bottom of the function.  As a matter of fact, since you've confirmed the problem affects more than one sound, you should make a `game_restart_looped_sounds` function and add this code to that function.  Then, when you do the engine wash fix, you can put that in the same place.

Reply:
I don't see the reasoning for starting the subspace sound when the mission is being ended.  It should only be restarted when the mission is going to be resumed, although the code will stop it again later, it's better for us not to have to depend on that or make the engine more inefficient in that way.

Unless there's something else I am not understanding?

Reply:
I agree, if the game is being ended then there isn't really a need to restart the sound.  However, the game isn't ended *immediately* -- it waits for the state transition.

What it boils down to is that `game_start_time()` is at the bottom, so the sound restart should also be at the bottom.  It makes the function symmetrical with the time and sound functions up top.

Reply:
From a human perspective, it does end immediately.  The game takes what, one, two frames to change states?  This is an imperceptible amount of time, and instead of allowing that transition to happen naturally, we are going to _call attention to it_ by restarting the sound?

For the user, the mission has ended as soon as they press the button, and it is most natural for them to assume that the game is spending all of its time tearing everything down.  In fact, if the sound restarts, _it will feel unpolished_ because then the player is going to start thinking about the code themselves, asking "why did the sound come back?"  and breaking immersion..

Symmetry should not be a distinct objective in code writing.  Literally every other concern should come first.  This function is not hard to read.  The way that this patch is now even makes it easier to read because now it's pretty obvious which choice restarts the game.

If you can't have it any other way, maybe you should just submit your own pr.

Reply:
Fair enough.